components/brave_new_tab_ui/api/brave_news/feedListener.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
+import getBraveNewsController from '.'
+import {
+  BraveNewsControllerRemote,
+  FeedListenerInterface,
+  FeedListenerReceiver
+} from '../../../../../out/Component/gen/brave/components/brave_today/common/brave_news.mojom.m'
+
+export const addFeedListener = (listener: (feedHash: string) => void) =>
+  new (class implements FeedListenerInterface {
```

Comment:
Wow, didn't know that there's a class expression like this 👍

Reply:
Yeah, it's an anonymous class. JavaScript is full of funky features like this.

One that terrifies me is nested destructuring
```js
let { foo: { bar: { hello: result = 7 } } } = { foo: { bar: { hello: 'world' } } }
```